src/Scientist/Internals/GCHelper.cs

```diff
@@ -0,0 +1,132 @@
+    {
+        private const string DebugConfigurationName = "DEBUG";
+        internal const string ReleaseConfigurationName = "RELEASE";
+        internal const string Unknown = "?";
+
+        public static bool IsMono { get; } = Type.GetType("Mono.Runtime") != null;
+
+        public static bool IsFullFramework =>
+#if (NET451)
+            true;
+#else
+            System.Runtime.InteropServices.RuntimeInformation.FrameworkDescription.StartsWith(".NET Framework", StringComparison.OrdinalIgnoreCase);
+#endif
+
+        public static bool IsNetCore =>
+#if (NET451)
+            false;
+#else
+            System.Runtime.InteropServices.RuntimeInformation.FrameworkDescription.StartsWith(".NET Core", StringComparison.OrdinalIgnoreCase);
+#endif
+
+        private static long AllocationQuantum { get; } = CalculateAllocationQuantumSize();
+
+        private static readonly Func<long> GetAllocatedBytesForCurrentThreadDelegate = GetAllocatedBytesForCurrentThread();
+
+        /// <summary>
+        /// returns total allocated bytes (not per operation)
+        /// </summary>
+        /// <param name="excludeAllocationQuantumSideEffects">Allocation quantum can affecting some of our nano-benchmarks in non-deterministic way.
+        /// when this parameter is set to true and the number of all allocated bytes is less or equal AQ, we ignore AQ and put 0 to the results</param>
+        /// <returns></returns>
+        public static long GetAllocatedBytes()
+        {
+            var allocation = GetAllocatedBytesByRuntime();
+
+            if (IsNetCore)
+            {
+                return allocation;
+            }
+
+            return allocation <= AllocationQuantum ? 0L : allocation;
+        }
+
+        private static long GetAllocatedBytesByRuntime()
+        {
+            // Monitoring is not available in Mono, see http://stackoverflow.com/questions/40234948/how-to-get-the-number-of-allocated-bytes-
+            if (IsMono)
+            {
+                return 0;
+            }
+            //GC.Collect();
+
+#if (NET451)
+            // "This instance Int64 property returns the number of bytes that have been allocated by a specific 
+            // AppDomain. The number is accurate as of the last garbage collection." - CLR via C#
+            // so we enforce GC.Collect here just to make sure we get accurate results
+            return AppDomain.CurrentDomain.MonitoringTotalAllocatedMemorySize;
+#elif (NETSTANDARD2_0)
+            // it can be a .NET app consuming our .NET Standard package
+            if (IsFullFramework) 
+            {
+                GC.Collect();
+                return AppDomain.CurrentDomain.MonitoringTotalAllocatedMemorySize;
+            }
+            // https://apisof.net/catalog/System.GC.GetAllocatedBytesForCurrentThread() is not part of the .NET Standard, so we use reflection to call it..
+            return GetAllocatedBytesForCurrentThreadDelegate.Invoke();
+#elif (NETCOREAPP2_1)
```

Comment:
Is this for future use? The library doesn't target `netcoreapp2.1`, so this would never be true at compile-time. That said, it's probably best that the library never targets the app TFMs, so it'd be `NETSTANDARD2_1` (or whatever the next standard version becomes), so until that arrives this `#elif` is redundant.

Reply:
So I used this in development stage so technically it could be removed. I'd suggest keeping it unless it causes a problem though. Might also be useful in the future if people want ILMerge/Native compilation support.